packages/teleport/src/lib/tdp/client.ts

```diff
@@ -77,30 +79,56 @@ export default class Client extends EventEmitter {
   }
 
   processMessage(buffer: ArrayBuffer) {
-    const messageType = this.codec.decodeMessageType(buffer);
+    const messageType = this.codec._decodeMessageType(buffer);
     try {
       if (messageType === MessageType.PNG_FRAME) {
-        this.processFrame(buffer);
+        this.handlePngFrame(buffer);
+      } else if (messageType === MessageType.CLIENT_SCREEN_SPEC) {
+        this.handleClientScreenSpec(buffer);
+      } else if (messageType === MessageType.MOUSE_BUTTON) {
+        this.handleMouseButton(buffer);
+      } else if (messageType === MessageType.MOUSE_MOVE) {
+        this.handleMouseMove(buffer);
       } else if (messageType === MessageType.ERROR) {
         this.handleError(new Error(this.codec.decodeErrorMessage(buffer)));
       } else {
-        this.handleError(
-          new Error(`recieved unsupported message type ${messageType}`)
-        );
+        this.logger.warn(`received unsupported message type ${messageType}`);
```

Comment:
why did we decide not to `this.handleError`? won't it be confusing if nothing happens? same as the handleXXX funcs below (not entirely sure how it gets invoked)

Reply:
It's a DX +1 while giving nothing important up UX-wise:

Those cases should never happen, and only would show up if we screwed something up on the backend. Since its not super important that the `client[Player]` stop if an unexpected TDP message is received, I decided to just log it instead since it makes the development of new features easier (Zac asked for this change because he's developing the Clipboard feature).